### PR TITLE
Temperature

### PR DESCRIPTION
**Pull request by superjax**
Added Temperature support on IMU message
Added Custom ISR

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ OPTIONS		?=
 DEBUG ?=
 
 # Serial port/device for flashing
-SERIAL_DEVICE	?= /dev/ttyUSB0
+SERIAL_DEVICE	?= /dev/ttyUSB1
```
Comment (superjax) [inline Makefile]: This shouldn't have been added

File: src/sensors.c [hunk 2]
```diff
@@ -11,34 +12,50 @@ int16_t _accel_data[3];
...
     mpu6050_read_accel(_accel_data);
     mpu6050_read_gyro(_gyro_data);
-    correct_imu();
+    mpu6050_read_temperature(&_imu_temperature);
+
+    // correct according to known biases and temperature compensation
+    _accel_data[0] -= (_params.values[PARAM_ACC_X_TEMP_COMP]*_imu_temperature)/1000 + _params.values[PARAM_ACC_X_BIAS];
+    _accel_data[1] -= (_params.values[PARAM_ACC_Y_TEMP_COMP]*_imu_temperature)/1000 + _params.values[PARAM_ACC_Y_BIAS];
+    _accel_data[2] -= (_params.values[PARAM_ACC_Z_TEMP_COMP]*_imu_temperature)/1000 + _params.values[PARAM_ACC_X_BIAS];
+    _gyro_data[0] += _params.values[PARAM_GYRO_X_BIAS];
```
Comment (dpkoch) [inline src/sensors.c]: Should be a minus sign? (subtract the bias off)

Comment (superjax) [inline src/sensors.c]: Yep!  You're right.

File: src/main.c [hunk 3]
```diff
@@ -20,12 +20,17 @@
...
+serialPort_t *Serial1;
+
 int main(void)
 {
   // Configure clock, this figures out HSE for hardware autodetect
   SetSysClock(0);
   systemInit();
 
+  // Initialize Serial ports
+  Serial1 = uartOpen(USART1, NULL, 921600, MODE_RXTX);
```
Comment (dpkoch) [inline src/main.c]: So it looks this removes support for setting the baud rate via a param. Just checking, is that what you intended?

Comment (superjax) [inline src/main.c]: Oh.  Shoot.  no, that's not what I meant to do.

**Comment by dpkoch** [comment]
Cool, I think we're close here.

Here's how I envision the data flow happening:

1. When we do a bias calibration, the result gets stored in the params like you're doing now
2. When we run `update_imu`, we apply the stored accel bias and temperature compensation (which you're doing already) and the stored gyro biases (which isn't happening right now) to the values stored in `_accel_data` and `_gyro_data`
3. The estimator uses those compensated values as its inputs, but then also internally keeps track of an additional bias term to further correct it. Those bias terms get stored in a global variable, but not in the parameter value; that stays where it was from the calibration.
4. For the mavlink streaming, by default we stream the values corrected by the param stuff (already stored in the global `_accel_data`, `_gyro_data`). If we choose to apply the estimator corrections, then we additionally subtract those biases off.

Does that seem reasonable/in line with what you're thinking?

**Comment by superjax** [comment]
I think that'll work.  Sure, let's do that.  We'll have an adaptive gyro bias calculation, and the static bias.  Cool!

**Comment by superjax** [comment]
What do you think of it now?

**Comment by dpkoch** [comment]
Sweet! Looks great. I'm just going to test it with fcu_io and run the style script real quick, then I think we're good to go!

**Comment by dpkoch** [comment]
Going along with BYU-MAGICC/fcu_io2#24, let's get this guy wrapped up! Again, we still just need to decide if we want to go with floating point, in which case we'd merge in the temperature_float branch.

**Comment by dpkoch** [comment]
This pull request will close #32, #35, and #37

**Comment by superjax** [comment]
Alright, I'm going for it!

**Pull request merged**